android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java

```diff
@@ -78,45 +79,70 @@ public TiUIText(final TiViewProxy proxy, boolean field)
 
 		this.field = field;
 
-		int tvId;
-		try {
-			tvId = TiRHelper.getResource("layout.titanium_ui_edittext");
-		} catch (ResourceNotFoundException e) {
-			if (Log.isDebugModeEnabled()) {
-				Log.e(TAG, "XML resources could not be found!!!");
-			}
-			return;
+		int borderStyle = UIModule.INPUT_BORDERSTYLE_FILLED;
+		borderStyle = TiConvert.toInt(proxy.getProperty(TiC.PROPERTY_BORDER_STYLE), borderStyle);
+		switch (borderStyle) {
+			case UIModule.INPUT_BORDERSTYLE_BEZEL:
+			case UIModule.INPUT_BORDERSTYLE_LINE:
+			case UIModule.INPUT_BORDERSTYLE_ROUNDED:
+				textInputLayout = new TextInputLayout(new ContextThemeWrapper(
+					proxy.getActivity(), R.style.Widget_MaterialComponents_TextInputLayout_OutlinedBox));
+				textInputLayout.setBoxBackgroundMode(TextInputLayout.BOX_BACKGROUND_OUTLINE);
+				textInputLayout.setBoxBackgroundColor(Color.TRANSPARENT);
+				if (borderStyle == UIModule.INPUT_BORDERSTYLE_ROUNDED) {
+					float radius = (new TiDimension("5dp", TiDimension.TYPE_LEFT)).getAsPixels(textInputLayout);
+					textInputLayout.setBoxCornerRadii(radius, radius, radius, radius);
+				} else {
+					textInputLayout.setBoxCornerRadii(0, 0, 0, 0);
+				}
+				break;
+			case UIModule.INPUT_BORDERSTYLE_NONE:
+				textInputLayout = new TextInputLayout(new ContextThemeWrapper(
+					proxy.getActivity(), R.style.Widget_MaterialComponents_TextInputLayout_OutlinedBox));
+				textInputLayout.setBoxBackgroundMode(TextInputLayout.BOX_BACKGROUND_NONE);
+				break;
+			case UIModule.INPUT_BORDERSTYLE_UNDERLINED:
+				textInputLayout = new TextInputLayout(proxy.getActivity());
+				textInputLayout.setBoxBackgroundMode(TextInputLayout.BOX_BACKGROUND_NONE);
```

Comment:
Is this a typo and should be `BOX_BACKGROUND_OUTLINE` or is it on purpose?

Reply:
Don't worry. This is implemented correctly. The "underline" style provides the old behavior where the background needs to be transparent (hence the NONE) and only the underline is drawn.